client/src/components/TripElement/TripEntry.js

```diff
@@ -0,0 +1,146 @@
+  const isSelected =
+    activeTripId === id && appView !== AppView.NEW_TRIP ? '' : 'hidden'
+  const [isRenaming, setIsRenaming] = useState(false)
+  const [tripName, setTripName] = useState(trip.tripName)
+
+  const inputRef = useRef(null)
+
+  const handleInputChange = (event) => {
+    setTripName(event.target.value)
+  }
+
+  const handleCheckClick = () => {
+    dispatch(editTrip({ id, tripName }))
+    setIsRenaming(false)
+  }
+
+  const handleCancelClick = () => {
+    setTripName(trip.tripName)
+    setIsRenaming(false)
+  }
+
+  useEffect(() => {
+    if (isSelected === 'hidden') {
+      setIsRenaming(false)
+      setTripName(trip.tripName)
+    }
+  }, [isSelected, trip, appView])
+
+  return (
+    <div className='group relative'>
+      <Button
+        key={`sidebar-trip-entry-${id}`}
+        onClick={() => {
+          dispatch(setActiveTripId(id))
+          // TODO: remove after connected to backend, it is here to prevent error
+          if (
+            !isNaN(trip.destinationLongitude) ||
+            !isNaN(trip.destinationLatitude)
+          ) {
+            dispatch(
+              changeCoordinatesAndZoom({
+                longitude: trip.destinationLongitude,
+                latitude: trip.destinationLatitude,
+                zoom: ZOOM_CITY_LEVEL,
+              }),
+            )
+            dispatch(
+              clearAllMarkersAndAdd_Store([
+                {
+                  longitude: trip.destinationLongitude,
+                  latitude: trip.destinationLatitude,
+                },
+              ]),
+            )
+          } else {
+            console.log(`no coordinate yet`)
+          }
+        }}
+        className={buttonClassName}
+      >
+        {!isRenaming ? (
+          trip.tripName
+        ) : (
+          <input
+            ref={inputRef}
+            className='items-center bg-green-200/40 font-medium'
+            placeholder={trip.tripName}
+            value={tripName}
+            onChange={handleInputChange}
+          />
+        )}
+      </Button>
+      <div className='absolute right-0 top-0 flex flex-row'>
+        {!isRenaming ? (
+          <>
+            <Button
+              key={`sidebar-trip-button-${id}-rename`}
+              onClick={() => {
+                setIsRenaming(true)
+              }}
+              className={`${isSelected} px-0 hover:text-red-400`}
+            >
+              <EditOutlinedIcon className='align-baseline' />
+            </Button>
+            <Button
+              key={`sidebar-trip-button-${id}-edit`}
+              onClick={() => {
+                dispatch(closeSidebar())
+                dispatch(openEditTripModal())
+              }}
+              className={`${isSelected} hover:text-red-400`}
+            >
+              <SettingsOutlinedIcon className='align-baseline' />
+            </Button>
+          </>
+        ) : (
+          <>
+            <Button
```

Comment:
![image](https://github.com/cpsc455-bugstorm/TravelersTea/assets/30674558/4c55630d-eec0-4647-92e4-d6f3b64b4dac)

Would you know why the check is inside the box? I think it might be nicer moved out, tho looking at the code idk why it's inside

Reply:
it is positioned absolute and it got small enough for the form to overlap. i was just trying to complete the functionality not keeping screensizes in mind

Reply:
thats all good then